src/Spd.Utilities.Dynamics/DynamicsContextLookupHelpers.cs

```diff
@@ -429,5 +429,16 @@ public static string LookupLicenceCategoryKey(Guid? licenceCategoryId)
                     throw;
             }
         }
+        public static async Task<contact?> GetDuplicateContact(this DynamicsContext context, contact contact, CancellationToken ct)
+        {
+            if (string.IsNullOrEmpty(contact.spd_bcdriverslicense))
+                return null;
+
+            return await context.contacts
+                .Where(a => a.firstname == contact.firstname &&
+                       a.lastname == contact.lastname &&
+                       a.birthdate == contact.birthdate &&
+                       a.spd_bcdriverslicense == contact.spd_bcdriverslicense).SingleOrDefaultAsync(ct);

```

Comment:
Returns the only element of a sequence, or a default value if the sequence is empty; this method throws an exception if there is more than one element in the sequence -singleOrDefault
We do not want user failed in submitting application if our data has problem. 
So,please use firstOrDefault

Reply:
And I think you need to do the same thing in CM CRC application